application.py

```diff
@@ -1,23 +1,81 @@
 from .baseapi import BaseAPI
+from .resources import Message
+from .device import Device
 
 class Application(BaseAPI):
 
+    def __init__(self, token=None):
+        super(Application, self).__init__(token)
+        self.token = token
+        self.device = Device(self.token)
+
     def get_all(self):
-        return self.request('application', 'GET')
+        return self.request('application', 'GET')['d']
+
+    def get(self, name):
+        params = {
+            'filter': 'app_name',
+            'eq': name
+        }
+        app = self.request('application', 'GET', params=params)['d']
+        if app:
+            return app
+        else:
+            print(Message.NO_APPLICATION_FOUND.format(value=name,app_att="name"))
+
+    def has(self, name):
+        params = {
+            'filter': 'app_name',
+            'eq': name
+        }
+        app = self.request('application', 'GET', params=params)['d']
+        if app:
+            return True
+        else:
+            return False
 
-    def restart(self, app_id):
-        return self.request('/application/' + str(app_id) + '/restart', 'POST')
+    def has_any(self):
+        apps = self.request('application', 'GET')['d']
+        if apps:
+            return True
+        else:
+            return False
 
     def get_by_id(self, app_id):
         params = {
             'filter': 'id',
             'eq': app_id
         }
-        try:
-            return self.request('application', 'GET', params=params)
-        except IndexError:
+        app = self.request('application', 'GET', params=params)['d']
+        if app:
+            return app
+        else:
             # found no application
-            print(Message.NO_APPLICATION_FOUND.format(value=app_id,dev_att="id"))
-        except:
-            # unexpected exception
-            raise
+            print(Message.NO_APPLICATION_FOUND.format(value=app_id,app_att="id"))
+
+    def create(self, name, device_type):
+        device_slug = self.device.get_device_slug(device_type)
+        if device_slug:
+            data = {
+                'app_name': name,
+                'device_type': device_slug[0]
```

Comment:
I would expect that ``self.device.get_device_slug(device_type)`` would give me a string. Why not?